### PR TITLE
Filter out unparsable projects when smart matching

### LINKED ISSUE #37: Smart matching is broken if projects can't be parsed
See console dump below. I haven't investigated the cause yet, I assume it's because we fail if we encounter something we can't parse, when we should probably just skip it.

```console
jmapple@Justins-MacBook-Pro p1-insta485-static % agio submissions -c eecs485 -p p1 -g jmapple last --download
Error: unsupported assignment type: 'testing'.  Recognized shortcuts: p, proj, project, l, lab, h, hw, hwk, homework
jmapple@Justins-MacBook-Pro p1-insta485-static % agio projects -c eecs485 --list
[1527] Project 1 - Templated Static Site Generator
[1525] Project 2 - Server-side Dynamic Pages
[1524] Project 3 - Client-side Dynamic Pages
[1526] Project 4 - MapReduce
[1523] Project 5 - Search Engine
[1749] Testing JVM errors 20.04
[1748] Testing JVM errors 22.04
```

### PR DESCRIPTION
**Pull request by japplefield**
Closes #37. 

Probably should also add some tests for this case.

### CODE REVIEW

File: agiocli/utils.py [hunk 1]
```diff
@@ -303,6 +303,7 @@ def parse_project_string(user_input):
     asstype_abbrev = match.group("asstype").lower()
     if asstype_abbrev not in assignment_types:
         asstypes = ", ".join(assignment_types.keys())
+        return False
         sys.exit(
             f"Error: unsupported assignment type: '{asstype_abbrev}'.  "
             f"Recognized shortcuts: {asstypes}"
```
Comment (japplefield) [inline agiocli/utils.py:309]: Not sure what to do about this error message now.

**Comment by japplefield** [comment]
No idea why CI is failing.

**Comment by noah-weingarden** [comment]
> No idea why CI is failing.

Pycodestyle error:

agiocli/utils.py:383:5: E303 too many blank lines (2)

**Comment by japplefield** [comment]
TY @noah-weingarden I was working on this in OH and didn't get to finish xd

File: agiocli/utils.py (hunk 1 above)
Comment (awdeorio) [reply agiocli/utils.py:309]: Should we throw an exception and catch it in the caller?

**Comment by japplefield** [comment]
```console
  ************* Module agiocli.utils
  agiocli/utils.py:235:31: E1123: Unexpected keyword argument 'options_map_func' in function call (unexpected-keyword-arg)
  agiocli/utils.py:402:28: E1123: Unexpected keyword argument 'options_map_func' in function call (unexpected-keyword-arg)
  agiocli/utils.py:596:31: E1123: Unexpected keyword argument 'options_map_func' in function call (unexpected-keyword-arg)
```

**Comment by japplefield** [comment]
image

File: agiocli/utils.py [hunk 2]
```diff
@@ -316,6 +320,14 @@ def parse_project_string(user_input):
     return asstype, num, subtitle
 
 
+def parse_project_string_filter(user_input):
```
Comment (awdeorio) [inline agiocli/utils.py:323]:
```suggestion
def parse_project_string_skip_errors(user_input):
```

Comment (awdeorio) [inline agiocli/utils.py:323]:
```suggestion
def parse_project_string_skiperr(user_input):
```

**Review by awdeorio — APPROVED** [review]
LGTM, just add a test please!

**Pull request merged**